dessia_common/document_generator.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+""" Write document file. """
+from typing import List, Tuple
+
+import docx
+
+
+class DocxWriter:
+    """ write a docx file. """
+
+    def __init__(self, filename: str):
+        self.filename = filename
+        self.doc = docx.Document()
+        self.headings = None
+
+    def add_headings(self, headings: List[Tuple[str, int]]) -> 'DocxWriter':
+        """ Add a list of headings to the document. """
+        new_doc = self.doc
+        for heading in headings:
+            new_doc.add_heading(heading[0], heading[1])
+        new_writer = DocxWriter(self.filename)
+        new_writer.doc = new_doc
+        new_writer.headings = headings
+        new_writer.add_page_breaks(1)
+        return new_writer
+
+    def add_paragraphs(self, paragraphs: List[str]) -> 'DocxWriter':
+        """ Add a list of paragraphs to the document. """
+        new_doc = self.doc
+        new_headings = self.headings
+        for paragraph in paragraphs:
+            if new_headings:
+                new_doc.add_heading(new_headings[0][0], new_headings[0][1])
```

Comment:
What is `heading[i][j]` ? It looks like it is an arbitrary structure. Should you create a class for heading ?